### PR TITLE
Implement flags grouping for karmada-aggregated-apiserver

### PR DESCRIPTION
**Pull request by AllenZMC**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Implement flags grouping for karmada-aggregated-apiserver #1485 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/aggregated-apiserver/app/aggregated-apiserver.go [hunk 1]
```diff
@@ -32,8 +36,21 @@ func NewAggregatedApiserverCommand(ctx context.Context) *cobra.Command {
 		},
 	}
 
-	opts.AddFlags(cmd.Flags())
+	fss := cliflag.NamedFlagSets{}
+
+	genericFlagSet := fss.FlagSet("generic")
+	genericFlagSet.AddGoFlagSet(flag.CommandLine)
```
Comment (lonelyCZ) [inline cmd/aggregated-apiserver/app/aggregated-apiserver.go:42]: I just tested it `genericFlagSet.AddGoFlagSet(flag.CommandLine)` that also not need to be added seemingly.

Comment (AllenZMC) [reply cmd/aggregated-apiserver/app/aggregated-apiserver.go:42]: @lonelyCZ , I'm curious about this, the result is the same with or without this sentence.
`// AddFlags adds flags to the specified FlagSet.
func (o *Options) AddFlags(flags *pflag.FlagSet) {
	o.RecommendedOptions.AddFlags(flags)
	flags.Lookup("kubeconfig").Usage = "Path to karmada control plane kubeconfig file."
`

Comment (lonelyCZ) [reply cmd/aggregated-apiserver/app/aggregated-apiserver.go:42]: Yes, so we don't need this sentence as we don't use `flag.CommandLine`.

Comment (AllenZMC) [reply cmd/aggregated-apiserver/app/aggregated-apiserver.go]: Solved @lonelyCZ

**Review by RainbowMango — APPROVED** [review]
Thanks guys. You give me enough confidence that I don't have to test again on my side.
/approve

**Pull request merged**